### DISCUSSION TITLE
Why is only the readme file being published to github pages

### DISCUSSION THREAD

**Discussion by ironmonkeeeee** [community/community#22400] [Other Feature Feedback, Questions, & Ideas]
<p>Hello,  I am struggling to figure out why the only thing I see on github pages is the readme file. </p>
<p>I have been giving myself a crash course on all this for the last week so excuse my ignorance</p>
<p>I believe that I have the repo set up properly here:</p>
<p><a href="https://github.com/ironmonkeeeee/ironmonkeeeee" rel="nofollow noopener">https://github.com/ironmonkeeeee/ironmonkeeeee</a></p>
<p>What I am trying to do is add google ad sense tags to the website that the above repo builds. In order to see if any changes I make are working and to give me a sandbox I followed the instructions to publish to a github page here:</p>
<p><a href="https://ironmonkeeeee.github.io/ironmonkeeeee/" rel="nofollow noopener">https://ironmonkeeeee.github.io/ironmonkeeeee/</a></p>
<p>However all I see is the readme file. Is this right? </p>
<p>Since this is a working website I was expecting to see a copy of <a href="http://www.eostalk.io" rel="nofollow noopener">www.eostalk.io</a> Am I completely missing something here?</p>
<p>Thank you in advance for your help!</p>
<p>-Dan</p>

**Comment by angela-crist** [comment, marked as answer]
<p>Hi <a class="mention" href="/ironmonkeeeee">@ironmonkeeeee</a>,</p>
<p>Looks like you’ve got your content in a folder, so your site is sort of available here: <a href="https://ironmonkeeeee.github.io/ironmonkeeeee/public/" rel="nofollow noopener">https://ironmonkeeeee.github.io/ironmonkeeeee/public/</a>.</p>
<p>If you want that to be your homepage, you will need to move it into the root of the repository instead of that folder. Then you should be all set! </p>
<p>Hope that helps!</p>

**Reply by MakiniMakini** [reply]
This worked for me, thank you

**Reply by an-normy** [reply]
thank god

**Reply by julie-mac** [reply]
I'm having the same issue! I moved my index.html file to the root of the repo but it's still not working :( Does it take a while to update? Also, should I unpublish then redeploy the site?

**Reply by ghost** [reply]
my file is in the root, but it can't showing

**Reply by leonaAtkins** [reply]
There also appears to be a weird quirk where, if your site is hosted at root (ours uses middleman, don't ask), on first build it shows the README.md rather than the site, but if the actions are re-ran, it works, very very odd and can only put it down to a github quirk.

**Comment by ironmonkeeeee** [comment]
<p>Thanks! I actually was able to get that far in all of my tinkering around. Maybe I am just too unskilled to get the full page up there. In all of the tutorials I watched it shows that the index.md file is what was being used to make web pages. I guess my quest now is to find out what other files need to be there as well. I am trying to see where the ad slots will be and how they will display on the page when fully rendered.</p>
<p>Regards,</p>
<p>Dan</p>

**Comment by nicks9188** [comment]
<p>What is the issue in this : <a href="https://github.com/nicks9188/static" rel="nofollow noopener">https://github.com/nicks9188/static</a></p>
<p>this is the url: <a href="https://nicks9188.github.io/static/" rel="nofollow noopener">https://nicks9188.github.io/static/</a></p>
<p>another one: <a href="https://github.com/nicks9188/github.io-self" rel="nofollow noopener">https://github.com/nicks9188/github.io-self</a></p>
<p>url: <a href="https://nicks9188.github.io/github.io-self/" rel="nofollow noopener">https://nicks9188.github.io/github.io-self/</a></p>

**Comment by rbulph** [comment]
<p>Same problem here. I have an index file which I was expecting to be displayed with github pages. But instead just the readme file is shown:</p>
<p><a href="https://rbulph.github.io/resipsa/#/" rel="nofollow noopener">https://rbulph.github.io/resipsa/#/</a></p>
<p><a href="https://github.com/rbulph/resipsa" rel="nofollow noopener">https://github.com/rbulph/resipsa</a></p>
<p>The index file is in the master folder, not a subfolder.</p>
<p>How do I fix this?</p>
<p>Presumably being able to view what you are developing through github is fundamental to the service. Otherwise everyone working on the project has to keep downloading the files, testing them out locally, and then uploading them again. Which would be very slow. So it’s a shame this hasn’t worked more smoothly.</p>

**Comment by rbulph** [comment]
<p>Also, assuming I can get this to work, how do I get it to work in relation to the branch that I am working on? I have created a new branch for my project, but the only options that I get for github pages are master branch, master branch/docs folder or none. My new branch doesn’t appear.</p>

**Comment by t-gottschalk** [comment]
<p>Hi,</p>
<p>I’m not exactly following you and I’m having the same issue where my project redirects to the Readme file. When I deploy to github pages using ssh and NPM ‘gh-pages’ it wants me to run an ‘npm run deploy’ which runs a build script and places files in a build folder. So now there is a ‘build’ folder, a ‘public’ folder and with React a ‘src’ folder as well. So therefore I am able to access this:</p>
<p><a href="https://t-gottschalk.github.io/nfl-team-rosters/index.html" rel="nofollow noopener">https://t-gottschalk.github.io/nfl-team-rosters/index.html</a></p>
<p>but not this:</p>
<p><a href="https://t-gottschalk.github.io/nfl-team-rosters/index.html" rel="nofollow noopener">https://t-gottschalk.github.io/nfl-team-rosters</a></p>
<p>Please advise on what I need to do to fix. Thank you for your time.</p>
<p>Best,</p>
<p>Tony</p>

**Reply by Anthonia-alexis** [reply]
I'm having the same issue too. Please, any help?

**Reply by ChadLaBombarde** [reply]
I think learning javascript was easier than GitHub

**Comment by ghost** [comment]
<p>hi the same problem happened to me and i dont know how to fix it can somebody pleas help me</p>

**Comment by CuriousCocainist** [comment]
<p>Hi,</p>
<p>It looks like they forgo to mention that you most probably shouldn’t initiate that repo with .md file…</p>
<p>Webpage seems to delete fine after deleting it :slight_smile:</p>

**Comment by daryllman** [comment]
<p>Faced the same problem and summarised all my findings over here, may be a fix for you:</p>
<p><a href="https://medium.com/@daryllwong/github-pages-does-not-work-for-me-f9d48fd44a5f" rel="nofollow noopener">https://medium.com/@daryllwong/github-pages-does-not-work-for-me-f9d48fd44a5f</a></p>

**Comment by smcaleese** [comment]
<p>I ran <code>npm run deploy</code> and had the same problem - GitHub pages was only showing the README.</p>
<p>Then I pushed the build directory I had created locally to my remote repository. When I did that, my React app worked with GitHub pages.</p>

**Reply by nugrezo** [reply]
did not work fo me in the same way I did npm run build git add .  git commit -m "build pushed" git push origin main and then npm run deploy but nothing

**Comment by tajmone** [comment]
<h1>Static sites need <code>.nojekyll</code> file!</h1>
<p>If you’re planning to publish a static HTML website on GH Pages, you need to add a <code>.nojekyll</code> file to disable Jekyll, otherwise the default behavior is that GH Pages will serve your markdown docs via Jekyll.</p>
<p>Go the website branch or folder (depending on your GH Pages settings) and in the website root folder open the terminal and type</p>
<pre><code class="lang-auto">touch .nojekyll
</code></pre>
<p>(or whichever method you prefer to create an empty file with that name)</p>

**Reply by rohit-077** [reply]
Thank you sooooo much. I've spent the past hour or two trying out everybody's suggestions but little did i know that this small step would work like magic.

**Comment by danishfox** [comment]
<p>hi please help me to fine my photo<br>
best regards<br>
tom dalby<br>
ps my phone 81 28 57 00</p>

**Comment by ghost** [comment]
<p>Why does Github have this Jekyll feature? What is the benefit that Github expects?</p>

**Comment by tajmone** [comment]
<p>Jekyll is a blog-style CMS, with contents in markdown:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://jekyllrb.com/favicon.ico" class="site-icon" width="32" height="32">
      <a href="https://jekyllrb.com/" target="_blank" rel="noopener nofollow ugc">Jekyll • Simple, blog-aware, static sites</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/11064292/181070764-9406670b-98de-4c66-a904-7da5aacd822b.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9bb05134cb18fa7763a4c7cf4a020c5b9c7ce870_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9bb05134cb18fa7763a4c7cf4a020c5b9c7ce870_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/b/9bb05134cb18fa7763a4c7cf4a020c5b9c7ce870.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9bb05134cb18fa7763a4c7cf4a020c5b9c7ce870_2_10x10.png"></div>

<h3><a href="https://jekyllrb.com/" target="_blank" rel="noopener nofollow ugc">Jekyll • Simple, blog-aware, static sites</a></h3>

<p>Transform your plain text into static websites and blogs</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>The advantage of using Jekyll is that your markdown documents in the GH Pages folder (or branch) would be automatically served as a blog, you only have to chose a template and not worry about HTML and CSS, only the markdown contents.</p>
<p>If you prefer creating a static website (in HTML), then just add the <code>.nojekyll</code> file in the GH Pages folder and GitHub will not use Jekyll to treat your project/account website ad dynamic…</p>

**Comment by CaptainOfPhB** [comment]
<p>Works fine for me, thanks!</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="dev-ira" data-post="14" data-topic="10431" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 dev-ira:</div>
<blockquote>
<p>Why does Github have this Jekyll feature? What is the benefit that Github expects?</p>
</blockquote>
</aside>
<p>Jekyll is a static CMS, which turns your Markdown documents into a full fledged blog/website:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://jekyllrb.com">
  <header class="source">
      <img src="https://jekyllrb.com/favicon.ico" class="site-icon" width="32" height="32">

      <a href="https://jekyllrb.com" target="_blank" rel="noopener nofollow ugc">Jekyll • Simple, blog-aware, static sites</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/11064292/181070767-05014a3f-5495-4703-94a1-7ebfec49b746.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9bb05134cb18fa7763a4c7cf4a020c5b9c7ce870_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9bb05134cb18fa7763a4c7cf4a020c5b9c7ce870_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/b/9bb05134cb18fa7763a4c7cf4a020c5b9c7ce870.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/b/9bb05134cb18fa7763a4c7cf4a020c5b9c7ce870_2_10x10.png"></div>

<h3><a href="https://jekyllrb.com" target="_blank" rel="noopener nofollow ugc">Jekyll • Simple, blog-aware, static sites</a></h3>

  <p>Transform your plain text into static websites and blogs</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>For many GH users, Jekyll is the preferred solution because it allows them to setup a website quickly, separating contents from style (free themes can be chosen when using Jekyll). No need to worry about HTML tags, navigation bars, etc., Jekyll handles all of that from the markdown and settings files.</p>
<p>So, the <code>.nojekyll</code> file is needed to let GitHub Pages know that you’re going to serve a static HTML website instead, and that you won’t be needing Jekyll.</p>
